spec/lib/jobs/dell-wsman-config-spec.js

```diff
@@ -0,0 +1,124 @@
+            wsmanConfigJob._initJob();
+            return expect(wsmanConfigJob.dellConfigs.configServerPath).to.be.equal('/api/smi/uri');
+        });
+
+        it('should throw error if dell configuration is imcompleted', function() {
+            configuration.get.withArgs('dell').returns({});
+            return expect(function() {
+                wsmanConfigJob._initJob();
+            }).to.throw('Config server path for Dell web service is not defined in wsmanConfig.json.');
+        });
+    });
+
+    describe('_handleSyncRequest', function() {
+        it('should send request successfully', function() {
+            configuration.get.withArgs('dell').returns(dellConfigs);
+            wsmanConfigJob._initJob();
+            wsmanTool.prototype.clientRequest.resolves({body: 'test'});
+            return expect(wsmanConfigJob._handleSyncRequest()).to.be.fulfilled;
+        });
+    });
+
+    describe('_handleSyncResponse', function() {
+        it('should reject if response is invalid', function() {
+            return expect(wsmanConfigJob._handleSyncResponse({})).to.
+                be.rejectedWith('Response for wsman microservice configuration is invalid.');
+        });
+
+        it('should update micro service config successfully', function() {
+            var response = {
+                credentials: {
+                    username: 'test',
+                    password: 'test'
+                },
+                gateway: 'http://1.1.1.1',
+                service: [
+                    {
+                        name: 'firmware',
+                        endpoint: [
+                            {
+                                name: 'updater',
+                                url: '/api/1.0/server/firmware/updater'
+                            }
+                        ]
+                    }
+                ]
+            };
+            fs.writeFile.yields();
+            return expect(wsmanConfigJob._handleSyncResponse(response)).to.be.fulfilled;
+        });
+
+        it('should reject if error occurs while update config file', function() {
+            var response = {
+                credentials: {
+                    username: 'test',
+                    password: 'test'
+                },
+                gateway: 'http://1.1.1.1',
+                service: [
+                    {
+                        name: 'firmware',
+                        endpoint: [
+                            {
+                                name: 'updater',
+                                url: '/api/1.0/server/firmware/updater'
+                            }
+                        ]
+                    }
+                ]
+            };
+            fs.writeFile.yields(new Error('test'));
+            return expect(wsmanConfigJob._handleSyncResponse(response)).to.
+                be.rejectedWith('Could not write wsman microservice configs to file');
```

Comment:
I am not sure how this works, you put a callback into promise, suppose it is asynchronous.

Reply:
I guess yes. The "return" keyword is used to publish UT result asserted asynchronously, otherwise such case would always succeed.
http://chaijs.com/plugins/chai-as-promised/

Reply:
I got you, perhaps you should try stub.yieldsAsync().  stub.yields should be executed synchronously, that means you turned a callback that should be async into sync.

Reply:
And there is a obviously bug in the job that put async callback into a promise, it should be corrected.